src/arkode/arkode_interp.c

```diff
@@ -311,29 +312,18 @@ void arkInterpPrintMem_Hermite(ARKInterp interp, FILE* outfile)
   ---------------------------------------------------------------*/
 int arkInterpSetDegree_Hermite(ARKodeMem ark_mem, ARKInterp interp, int degree)
 {
-  /* if this degree is already stored, just return */
-  if (abs(degree) == HINT_DEGREE(interp)) { return (ARK_SUCCESS); }
-
-  /* on positive degree, check for allowable value and overwrite stored degree */
-  if (degree >= 0)
+  if (degree > ARK_INTERP_MAX_DEGREE)
   {
-    if (degree > ARK_INTERP_MAX_DEGREE)
-    {
-      arkProcessError(ark_mem, ARK_INTERP_FAIL, __LINE__, __func__, __FILE__,
-                      "Illegal degree specified.");
-      return (ARK_ILL_INPUT);
-    }
-
-    HINT_DEGREE(interp) = degree;
-    return (ARK_SUCCESS);
+    arkProcessError(ark_mem, ARK_INTERP_FAIL, __LINE__, __func__, __FILE__,
+                    "Illegal degree specified.");
+    return ARK_ILL_INPUT;
   }
 
-  /* on negative degree, check for allowable value and update stored degree */
-  degree = -degree;
-  if (degree > ARK_INTERP_MAX_DEGREE) { degree = ARK_INTERP_MAX_DEGREE; }
-  HINT_DEGREE(interp) = SUNMIN(HINT_DEGREE(interp), degree);
+  if (degree < 0) { degree = ARK_INTERP_MAX_DEGREE; }
```

Comment:
I don't think that this routine can be called with a negative input anymore, since you removed these calls from the steppers.  I'll need to think through the new logic you've put together to determine whether the previous behavior is fully replicated (I anticipate that it's fine, now that you added the `ark_mem->interp_degree` variable); however, since a negative input can no longer occur, please update the comment block at the top of this function, and remove this line.
```suggestion
```